ci(documentation-website): lighthouse check with nginx

itsmacr8 opened the pull request:
Replace nodejs lighthouse test with nginx because of poor performance score.

Closes #953

# The Pull Request is ready

- [ ] fixes #<!-- YOUR ISSUE ID HERE -->
- [ ] all actions are passing
- [ ] only fixes a single issue

## Overview

<!-- Provide a brief description of the changes introduced by this
Pull Request. -->

## Review points

<!-- List the points to be reviewed in detail 
and the points you are not confident about. -->
<!-- Delete this section if not needed -->

## Documentation-Website

- [ ] mobile view is usable
- [ ] desktop view is usable
- [ ] no a-tags are used directly (NavLink, MailLink, ExternalLink instead)
- [ ] all new texts are added to the translation files (at least the english one)
- [ ] tests have been added (if required)
- [ ] shared code has been extracted in a different file

## Notes

<!-- Write any note or comment. You can share your thoughts or ideas. -->
<!-- Delete this section if not needed -->

itsmacr8 committed:
ci(documentation-website): lighthouse check with nginx

Replace nodejs lighthouse test with nginx because of poor performance score.

Closes #953

Idrinth commented on nginx.conf:
```diff
@@ -0,0 +1,59 @@
+    default_type application/octet-stream;
+
+    # SSL Settings
+    ssl_protocols TLSv1 TLSv1.1 TLSv1.2; # Dropping SSLv3, ref: POODLE
+    ssl_prefer_server_ciphers on;
```
no ssl during the tests

itsmacr8 replied:
So, I have to remove these two lines and don't have to add anything, right?

Idrinth replied:
yes, anything with ssl won't be used

Idrinth commented on nginx.conf:
```diff
@@ -0,0 +1,59 @@
+    gzip_disable "msie6";
+
+    # Virtual Host Configs
+    include /etc/nginx/conf.d/*.conf;
+    include /etc/nginx/sites-enabled/*;
```
do we need those?

itsmacr8 replied:
So, I have to remove these two lines here also and don't have to add anything, right?

Idrinth replied:
I don't see a reason for them to exist here at least. You can see in the runner below if it works btw ;)

Idrinth commented on nginx.conf:
```diff
@@ -0,0 +1,59 @@
+        # Server name can be domain or IP
+        server_name idrinth-api-ben.ch;
+
+        location / {
+            # Proxy pass to the Node.js application
+            proxy_pass http://localhost:8080;
+            proxy_http_version 1.1;
+            proxy_set_header Upgrade $http_upgrade;
+            proxy_set_header Connection 'upgrade';
+            proxy_set_header Host $host;
+            proxy_cache_bypass $http_upgrade;
+        }
```
what are you proxying to? We don't have a nodejs server?

itsmacr8 replied:
I read lots of articles and then wrote accordingly that I understood because I had never done it before. So, what can we use in there?

Idrinth replied:
The whole block has to be removed, it doesn't do anything for us. We are serving static files, not a nodejs server.

Idrinth commented on nginx.conf:
```diff
@@ -0,0 +1,59 @@
+    server {
+        listen 80;
+        # Server name can be domain or IP
+        server_name idrinth-api-ben.ch;
```
this server will never be called, we are looking for localhost here - you are not leaving the machine.

itsmacr8 replied:
what can we do here? use 'http://localhost:8080'?

Idrinth replied:
you can stay on port 80, the server_name just has to be something you can actually use, 127.0.0.1 or localhost for example

Idrinth commented on .github/workflows/documentation-website.lighthouse-check.yml:
```diff
@@ -38,17 +36,14 @@ jobs:
+        sudo systemctl restart nginx
+    - name: Run Lighthouse Check
+      run: |
+        cd documentation-website && npm install
```
always npm ci in a pipeline, install updates unchecked

Idrinth replied:
also there is no subfolder

itsmacr8 replied:
Do we have to replace 'cd documentation-website && npm install' with 'npm ci'?

Idrinth replied:
yes

Idrinth committed:
Merge branch 'the-one' into documentation-website-issue-#953

itsmacr8 committed:
ci(documentation-website): lighthouse check with nginx

- Removed unnecessary settings from nginx file such as SSL Settings, Virtual Host Configs, Proxy pass to the Node.js application
- Update npm script from lighthouse-check file

Closes #953

Co-Authored-By: Björn Büttner <Idrinth@users.noreply.github.com>

itsmacr8 committed:
Merge branch 'documentation-website-issue-#953' of https://github.com/itsmacr8/documentation-website into documentation-website-issue-#953

itsmacr8 commented:
Let me know if more changes are needed, @Idrinth

Idrinth commented on nginx.conf:
```diff
@@ -0,0 +1,41 @@
+    gzip_disable "msie6";
+
+    # Reverse proxy configuration
+    server {
+        listen 80;
+        # Server name can be domain or IP
+        server_name localhost;
+    }
```
This doesn't serve anything - there needs to be a directory(dist) served

Idrinth replied:
```nginx
    root /home/runner/work/documentation-website/documentation-website/dist;
    index index.html;

    location / {
        add_header Cache-Control no-cache;
        expires 0;
        try_files $uri $uri/ /index.html;
    }
```
for example

itsmacr8 replied:
Can I use it as it is or need some modification?

Idrinth replied:
can be used as is I believe

Idrinth commented on .github/workflows/documentation-website.lighthouse-check.yml:
```diff
@@ -38,17 +36,14 @@ jobs:
+    - name: Run Lighthouse Check
+      run: |
+        npm ci
+        npm run lighthouse
```
this script has not yet been adapted, it will still start a server of it's own

itsmacr8 replied:
What should I do here?

Idrinth replied:
adjust the script

itsmacr8 replied:
I have checked the package file, and there is `"lighthouse": "npm run build && start-server-and-test serve http://localhost:8080 'node tools/lighthouse-report.js'"` Isn't it adapted yet, and if not, which script is compatible now?

Idrinth replied:
the start server part has to be removed, only building and running the tools/lighthouse-report can stay.

Idrinth reviewed (changes requested):
There is a merge conflict as well, please fix that :)

itsmacr8 committed:
Merge remote-tracking branch 'upstream/the-one' into documentation-website-issue-#953

itsmacr8 committed:
ci(documentation-website): lighthouse check with nginx

update lighthouse npm script
update nginx server configuration

Idrinth commented on .github/workflows/documentation-website.lighthouse-check.yml:
```diff
@@ -0,0 +1,49 @@
+    branches:
+    - the-one
+    paths:
+      - 'documentation-website/**.ts'
+      - 'documentation-website/**.tsx'
+      - 'documentation-website/**.js'
+      - 'documentation-website/**.cjs'
+      - 'documentation-website/package-lock.json'
+      - '.github/workflows/documentation-website.lighthouse-check.yml'
+  pull_request:
+    branches:
+    - the-one
+    paths:
+      - 'documentation-website/**.ts'
+      - 'documentation-website/**.tsx'
+      - 'documentation-website/**.js'
+      - 'documentation-website/**.cjs'
+      - 'documentation-website/package-lock.json'
+      - '.github/workflows/documentation-website.lighthouse-check.yml'
```
why is that pointing to `documentation-website`? That folder exists nowhere
